src/openzaak/utils/views.py

```diff
@@ -113,3 +128,162 @@ def initialize_request(self, request, *args, **kwargs):
+            )
+
+        import_instance = Import.objects.create(
+            status=ImportStatusChoices.pending,
+            import_type=self.import_type,
+            # This should be updated when the import metadata file is parsed
+            total=0,
+        )
+
+        return Response(
+            data={
+                "upload_url": import_instance.get_upload_url(request=request),
+                "status_url": import_instance.get_status_url(request=request),
+                "report_url": import_instance.get_report_url(request=request),
+            },
+            status=status.HTTP_201_CREATED,
+            content_type="application/json",
+        )
+
+
+class CSVParser(BaseParser):
+    media_type = "text/csv"
+
+    def parse(self, stream, media_type=None, parser_context=None):
+        parser_context = parser_context or {}
+        encoding = parser_context.get("encoding", settings.DEFAULT_CHARSET)
+
+        try:
+            raw_data = stream.read()
+            data = raw_data.decode(encoding)
+            csv_reader = csv.reader(data)
+            len([row for row in csv_reader])
+        except (ValueError, csv.Error) as e:
+            raise ParseError(_("Unable to parse CSV file: %s") % force_str(e))
+
+        return data
+
+
+def validate_headers(import_data: StringIO, expected_headers: list[str]) -> None:
+    csv_reader = csv.reader(import_data, delimiter=",")
+
+    headers = next(csv_reader, [])
+    missing_headers = [
+        expected_header
+        for expected_header in expected_headers
+        if expected_header not in headers
+    ]
+
+    if missing_headers:
+        error_message = _(
+            "The following headers were not found in the CSV file: %s"
+        ) % ",".join(missing_headers)
+
+        raise ValidationError(
+            {"__all__": [error_message]}, code="missing-import-headers"
+        )
+
+
+class ImportUploadView(mixins.CreateModelMixin, viewsets.GenericViewSet):
+    lookup_field: str = "uuid"
+    parser_classes = (CSVParser,)
+    permission_classes = (ImportAuthRequired,)
+
+    import_type: ImportTypeChoices
+    import_headers: list[str] = []
+
+    def get_queryset(self):
+        return Import.objects.filter(import_type=self.import_type)
+
+    def get_import_headers(self) -> list[str]:
+        return self.import_headers
+
+    def create(self, request, *args, **kwargs):
+        import_instance = self.get_object()
+
+        if import_instance.status != ImportStatusChoices.pending:
+            import_status = ImportStatusChoices(import_instance.status)
+            error_message = (
+                _(
+                    "Starting an import process is not possible due to the current "
+                    "status of the import: %s"
+                )
+                % import_status.label
+            )
+            raise ValidationError({"__all__": [error_message]}, code="invalid-status")
+
+        request_data = request.data or ""
```

Comment:
Is empty request body allowed?

Reply:
See `55b365e82119aa939c89e8fd245f7ad3312131ea`